.gitignore

```diff
@@ -39,3 +39,6 @@ Make.dep
 .dirlinks
 .vscode
 .DS_Store
+build
+.ccls-cache
+compile_commands.json
```

Comment:
why need add these file?

Reply:
Hi

- I added `build` because by CMake tools automatically detected `CMakeLists.txt` and tried to build it. While I turned it off, I added this here in case somebody faces this as well.
- `.ccls-cache` and `compile_commands.json` are files related to LSPs and I mentioned this [here](https://github.com/apache/nuttx/pull/11647#discussion_r1476971523).